library/network/test/y2firewall/firewalld_test.rb

```diff
@@ -218,6 +233,18 @@
       expect(firewalld.read).to eq(false)
     end
 
+    it "stores the list of available zone names" do
+      expect(firewalld.current_zone_names).to eq([])
+      firewalld.read
+      expect(firewalld.current_zone_names).to eq(known_zones)
+    end
+
+    it "stores the list of available service names" do
+      expect(firewalld.current_service_names).to eq([])
+      firewalld.read
+      expect(firewalld.current_service_names).to eq(known_services)
```

Comment:
np: You could use the `#change` method:

```ruby
expect { firewalld.read }.to change { firewall.current_service_names }.from([]).to(known_services)
```

But it is up to you, as for some people it might be less readable.